build_scripts/create_sysroot.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+# This script customizes the plutosdr-fw sysroot image to our needs.
+# NOTE: you only need this script if you want to rebuild the sysroot
+#       folder, i.e. after a new release of the plutosdr-fw!
+#       The script is automatically invoked by the other build scripts
+
+# Required plutosdr-fw version
+PLUTOSDR_FW_TAG="v0.32"
+
+
+SCRIPT_DIR=`pwd`/`dirname "$0"`
+BUILD_DIR=$SCRIPT_DIR/build
+PLUTOSDR_FW_DIR=$BUILD_DIR/plutosdr-fw
+SYSROOT_DIR=$BUILD_DIR/pluto-custom.sysroot
+SRC_DIR=$SCRIPT_DIR/..
+
+mkdir -p $BUILD_DIR
+cd $BUILD_DIR || exit 1
+mkdir -p .tmp_create_sysroot
+cd .tmp_create_sysroot || exit 1
+
+# Read current build status
+BUILD_STATUS=$BUILD_DIR/build_status
+touch $BUILD_STATUS
+source $BUILD_STATUS || exit 1
+
+if [[ $STATUS_DOWNLOAD_SYSROOT != "y" ]]
+then
+  echo "Could not locate existing sysroot dir. Downloading sources"
+  wget https://github.com/analogdevicesinc/plutosdr-fw/releases/download/$PLUTOSDR_FW_TAG/sysroot-$PLUTOSDR_FW_TAG.tar.gz
+  tar -xzf sysroot-$PLUTOSDR_FW_TAG.tar.gz || exit 1
+  mv staging pluto-custom.sysroot
+  mv pluto-custom.sysroot $BUILD_DIR/
+  echo "STATUS_DOWNLOAD_SYSROOT=y" >> $BUILD_STATUS
+else
+  echo "Found existing sysroot folder. Skip download."
+fi
+
+echo "Bulding libraries"
+
+if [[ $STATUS_SYSROOT_GOT_FFTW3F != "y" ]]
+then
+  echo "Downloading fftw3"
+  wget http://www.fftw.org/fftw-3.3.8.tar.gz
+  tar -xzf fftw-3.3.8.tar.gz
+
+  echo "Installing fftw3"
+  cd  fftw-3.3.8 || exit 1
+  ./configure arm --enable-float --enable-shared --host=arm-linux-gnueabihf --prefix="$SYSROOT_DIR/usr/" CC="arm-linux-gnueabihf-gcc --sysroot=$SYSROOT_DIR/"
+  make install || exit 1
+  echo "STATUS_SYSROOT_GOT_FFTW3F=y" >> $BUILD_STATUS
+else
+  echo "FFTW3 is already present"
+fi
+
+if [[ $STATUS_SYSROOT_GOT_LIBCONFIG != "y" ]]
+then
+  echo "Downloading libconfig"
+  wget https://hyperrealm.github.io/libconfig/dist/libconfig-1.7.2.tar.gz
+  tar -xzf libconfig-1.7.2.tar.gz
+  echo "Installing libconfig"
+  cd libconfig-1.7.2 || exit 1
+  ./configure --host=arm-linux-gnueabihf --prefix="$SYSROOT_DIR/usr/" CC="arm-linux-gnueabihf-gcc --sysroot=$SYSROOT_DIR/"
+  make install || exit 1
+  echo "STATUS_SYSROOT_GOT_LIBCONFIG=y" >> $BUILD_STATUS
+else
+  echo "Libconfig is already present"
+fi
+
+if [[ $STATUS_SYSROOT_GOT_LIBFEC != "y" ]]
+then
+  cd $SRC_DIR/libfec || exit 1
```

Comment:
Add a check if this submodule has actually been downloaded (or force a git submodule update -- init --recursive on start of this script) as it will currently not fail if the submodules have not been initialized.